lib/okta.go

```diff
@@ -85,44 +99,55 @@ func NewOktaClient(creds OktaCreds, oktaAwsSAMLUrl string, sessionCookie string)
 	}, nil
 }
 
-func (o *OktaClient) AuthenticateProfile(profileARN string, duration time.Duration) (sts.Credentials, string, error) {
+func (o *OktaClient) AuthenticateUser() error {
 	var oktaUserAuthn OktaUserAuthn
 
-	// Attempt to reuse session cookie
-	var assertion SAMLAssertion
-	err := o.Get("GET", o.OktaAwsSAMLUrl, nil, &assertion, "saml")
+	// Step 1 : Basic authentication
+
+	user := OktaUser{
+		Username: o.Username,
+		Password: o.Password,
+	}
+
+	payload, err := json.Marshal(user)
 	if err != nil {
-		log.Debug("Failed to reuse session token, starting flow from start")
-		// Step 1 : Basic authentication
-		user := OktaUser{
-			Username: o.Username,
-			Password: o.Password,
-		}
+		return err
+	}
 
-		payload, err := json.Marshal(user)
-		if err != nil {
-			return sts.Credentials{}, "", err
-		}
+	log.Debug("Step: 1")
+	err = o.Get("POST", "api/v1/authn", payload, &oktaUserAuthn, "json")
+	if err != nil {
+		return fmt.Errorf("Failed to authenticate with okta: %#v", err)
+	}
 
-		log.Debug("Step: 1")
-		err = o.Get("POST", "api/v1/authn", payload, &oktaUserAuthn, "json")
-		if err != nil {
-			return sts.Credentials{}, "", errors.New("Failed to authenticate with okta.  Please check that your credentials have been set correctly with `aws-okta add`")
+	o.UserAuth = &oktaUserAuthn
+
+	// Step 2 : Challenge MFA if needed
+	log.Debug("Step: 2")
+	if o.UserAuth.Status == "MFA_REQUIRED" {
+		log.Info("Requesting MFA. Please complete two-factor authentication with your second device")
+		if err = o.challengeMFA(); err != nil {
+			return err
 		}
+	}
 
-		o.UserAuth = &oktaUserAuthn
+	if o.UserAuth.SessionToken == "" {
+		return fmt.Errorf("authentication failed for %s", o.Username)
+	}
 
-		// Step 2 : Challenge MFA if needed
-		log.Debug("Step: 2")
-		if o.UserAuth.Status == "MFA_REQUIRED" {
-			log.Info("Requesting MFA")
-			if err = o.challengeMFA(); err != nil {
-				return sts.Credentials{}, "", err
-			}
-		}
+	return nil
+}
+
+func (o *OktaClient) AuthenticateProfile(profileARN string, duration time.Duration) (sts.Credentials, string, error) {
+
+	// Attempt to reuse session cookie
+	var assertion SAMLAssertion
+	err := o.Get("GET", o.OktaAwsSAMLUrl, nil, &assertion, "saml")
+	if err != nil {
+		log.Debug("Failed to reuse session token, starting flow from start")
 
-		if o.UserAuth.SessionToken == "" {
-			return sts.Credentials{}, "", fmt.Errorf("authentication failed for %s", o.Username)
+		if err := o.AuthenticateUser(); err != nil {
+			return sts.Credentials{}, "", err
 		}
 
 		// Step 3 : Get SAML Assertion and retrieve IAM Roles
```

Comment:
let's update and not specify step 3 since that doesn't make sense in context anymore